### ISSUE TITLE
Fix punycode deprecation warning

### ISSUE THREAD

**Issue by cozykitten** [naptha/tesseract.js#876]
**Tesseract.js version (version number for npm/GitHub release, or specific commit for repo)**
5.0.4

**Describe the bug**
When updating to node v21 this node warning comes up when using packages that rely on node's built in punycode module.
``(node:15352) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.``

This module can be traced back to an old version of ``tr46`` which itself is used in an old version of ``whatwg-url`` which itself is used in an old version of ``node-fetch``.
tr46 already fixed that issue and newer versions of whatwg-url don't have the same issue and as far as I can tell the latest node-fetch version doesn't even use whatwg-url in the first place.

Long story short, would you consider updating to a newer version of node-fetch?

**To Reproduce**
1. Update node to v21
2. run project using tesseract.js
3. See error

**Device Version:**
 - OS + Version: Windows 11 22H2
 - Node v21.6

**Comment by Balearica** [comment]
Thanks for looking into this.  I looked into whether it is possible to remove this warning message by bumping the version of `node-fetch`, however I do not believe it is.  Tesseract.js uses `v2` of `node-fetch`. 
1. According to [this issue](https://github.com/node-fetch/node-fetch/issues/1797), the depreciation warning is still an issue for the latest `v2` release of `node-fetch`
2. According to [this documentation](https://github.com/node-fetch/node-fetch/blob/HEAD/docs/v3-UPGRADE-GUIDE.md#converted-to-es-module), using `v3` of `node-fetch` would not be possible without adapting the project to use ESM modules rather than CommonJS
   1. The documentation states "We recommend you stay on v2 which is built with CommonJS unless you use ESM yourself. We will continue to publish critical bug fixes for it."

Therefore, I believe this warning will continue until either (1) there is `v2` release of `node-fetch` that resolves it or (2) Tesseract.js switches to ESM modules.  The latter will probably happen in the next major version, however is currently not a priority for me.

**Comment by cozykitten** [comment]
Thank you, next major version sounds good enough.
For now I fixed it manually by going down the module tree and modifying the old version of tr46 and whatwg-url directly to use the userland punycode module.

**Comment by Balearica** [comment]
Interestingly, it looks like a stable version of `fetch` has been added to Node.js before either (1) `node-fetch` releases a version of `v2` without this issue or (2) the Node.js community switches to ESM.  Therefore, I think the solution to this may just be to use `node-fetch` as a polyfill for old Node.js versions and tell people annoyed by the warning message to upgrade.

**Comment by Balearica** [comment]
Reopening as this issue is not yet resolved.  Will close once this has been patched.

**Comment by Balearica** [comment]
There is some conflicting information regarding when `fetch` was added to Node.js, which is likely attributable to LLMs being used in search results.  To summarize what I found:

1. `fetch` was added behind an experimental flag (`--experimental-fetch`) in Node.js `v17.5.0`
   - https://nodejs.org/en/blog/release/v17.5.0
3. `fetch` was included by default, however still flagged as `experimental`, in Node.js `v18.0.0`
   - https://nodejs.org/en/blog/release/v18.0.0
   - `fetch` can be disabled using `--no-experimental-fetch`
3. `fetch` was marked as `stable` in `v21.0.0`
   - https://nodejs.org/en/blog/release/v21.0.0
   - This is flagged as a documentation change, so presumably indicates that `fetch` was working as intended with previous versions.  

Tesseract.js currently supports Node.js `v14` and higher.  Therefore, it looks like the only versions that require `node-fetch` are `v14` (reached end-of-life on April 30, 2023) and `v16` (reached end-of-life on `September 11, 2023`).

**Comment by cozykitten** [comment]
Yea, I remember using ``fetch`` since v18 and never had an issue with it

**Comment by Balearica** [comment]
Closing as this was implemented in [v6.0.1](https://github.com/naptha/tesseract.js/releases/tag/v6.0.1), which is now live on npm.
